CsharpExtras/Api/CsharpExtrasApi.cs

```diff
@@ -22,14 +23,19 @@ public void SetLogger(ILogger logger)
             StaticLogManager.Logger = logger;
         }
 
+        public IListValuedDictionary<TKey, TVal> NewListValuedDictionary<TKey, TVal>()
+        {
+            return new ListValuedDictionaryImpl<TKey, TVal>();
+        }
+
         public IComparer<T> NewDescendingComparer<T>() => new DescendingComparer<T>();
 
         public IBijectionDictionary<TKey, TVal> NewBijectionDictionary<TKey, TVal>()
             => new BijectionDictionaryImpl<TKey, TVal>();
         public IRegexPatternDictionary<TVal> NewRegexPatternDictionary<TVal>()
             => new RegexPatternDictionaryImpl<TVal>();
-        public IMultiValueMap<TKey, TVal> NewMultiValueMap<TKey, TVal>()
-            => new MultiValueMapImpl<TKey, TVal>();
+        public ISetValuedDictionary<TKey, TVal> NewSetDictionary<TKey, TVal>()

```

Comment:
I would rename this to `NewSetValuedDictionary` to match the type and the naming of the method to get a new list valued dictionary.